src/CSharp/CodeCracker/Style/TaskNameAsyncAnalyzer.cs

```diff
@@ -0,0 +1,54 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+
+namespace CodeCracker.Style
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class TaskNameAsyncAnalyzer : DiagnosticAnalyzer
+    {
+        public const string DiagnosticId = "CC0061";
+        internal const string Title = "Async method can be terminating with 'Async' name.";
+        internal const string MessageFormat = "Change method name including 'Async'.";
+        internal const string Category = SupportedCategories.Style;
+        const string Description = "Async method can be terminating with 'Async' name.";
+
+        internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+            DiagnosticId,
+            Title,
+            MessageFormat,
+            Category,
+            DiagnosticSeverity.Info,
+            isEnabledByDefault: true,
+            description: Description,
+            helpLink: HelpLink.ForDiagnostic(DiagnosticId));
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get { return ImmutableArray.Create(Rule); } }
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.RegisterSyntaxNodeAction(AnalyzeClass, SyntaxKind.MethodDeclaration);
+        }
+
+        public Task<bool> xpto()
+        {
+            return null;
+        }
+
+        private static void AnalyzeClass(SyntaxNodeAnalysisContext context)
+        {
+            var invocationExpression = (MethodDeclarationSyntax)context.Node;
+
+            var nameSyntax = invocationExpression.ReturnType?.ToString();
+            if (nameSyntax == null || !nameSyntax.Contains("Task")) return;
```

Comment:
Don't check for the `Task` name, check for the type using semantic analysis. Remember that we don't have to return a `Task`, I can return anything that inherits from it, and that would not have the `Task` name. Actually, we could even work the awaitable pattern, but that would be too much at this point, looking into anything that inherits from `Task` or `Task` itself would be enough.